### PR TITLE
Introduce GH Actions

### PR DESCRIPTION
**Pull request by chrfritsch**
Make sure these boxes are checked before submitting your pull request - thank you!

- [ ] All coding styles are fulfilled. ([How to check for cs issues?](https://github.com/BurdaMagazinOrg/thunder-dev-tools/blob/master/README.md#code-style-guidelines))
- [ ] All tests are running locally. ([How to run the test?](https://github.com/thunder/thunder-distribution/blob/8.x-4.x/docs/development.md#how-to-run-the-tests))
- [ ] Necessary update hooks are provided.
- [ ] User roles have correct access for new introduced permission.
- [ ] Every thunder module has a README.md in its root. Follow [this guidelines](https://www.drupal.org/node/2181737), but we don't need every topic.
- [ ] Code is covered with well-balanced amount of inline comments.

If you are really awesome, then your feature is covered by additional tests. Well done!

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+          path: ~/.composer/cache
+          key: ${{ runner.os }}-composer-cache-${{ hashFiles('**/composer.json') }}
+          restore-keys: ${{ runner.os }}-composer-cache-
+
+      - name: Setup environment variables
+        run: |
+          echo "::add-path::$HOME/.composer/vendor/bin"
+
+      - name: Get build environment
+        run: composer global require thunder/drupal-testing:dev-feature/min-testing
```
Comment (dbosen) [inline .github/workflows/test.yml]: this branch is already merged

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+      - name: Install suggested dependencies
+        run: composer require "thunder/thunder_performance_measurement:dev-master" "thunder/testsite_builder:dev-master" "drupal/media_entity_generic:^1.0" "drupal/riddle_marketplace:^3.0-beta2" "drupal/nexx_integration:^3.0" "valiton/harbourmaster:~8.1" --no-update --no-progress --working-dir=/tmp/test/thunder/install
+
+      - name: Build the docroot
+        run: test-drupal-project build
+
+      - name: Test for deprecations
+        run: test-drupal-project deprecation
+        continue-on-error: true
```
Comment (dbosen) [inline .github/workflows/test.yml]: Does this work? The next step will probably not start, as long as this fails

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    - name: Start MySql service
+      run: |
+        sudo /etc/init.d/mysql start
+
+    - name: Setup environment variables
+      run: |
+        echo "::add-path::$HOME/.composer/vendor/bin"
+
+    - name: Get build environment
+      run: composer global require thunder/drupal-testing:dev-feature/min-testing
```
Comment (dbosen) [inline .github/workflows/test.yml]: can be changed as well

File: .github/workflows/test.yml (hunk 2 above)
Comment (chrfritsch) [reply .github/workflows/test.yml]:
```
jobs.<job_id>.steps.continue-on-error

Prevents a job from failing when a step fails. Set to true to allow a job to pass when this step fails.
```

File: .github/workflows/do-not-merge.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+jobs:
+  do-not-merge:
+
+    runs-on: ubuntu-latest
+
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'DO NOT MERGE') }}
+
+    steps:
+      - run: false
```
Comment (dbosen) [inline .github/workflows/do-not-merge.yml]: I am not sure about this. I think its more relevant to quickly see, if your branch fails or not. The existance of the tag should be enough to make us not merge it.

Comment (chrfritsch) [reply .github/workflows/do-not-merge.yml]: Will remove it

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+
+  test-min:
+
+    runs-on: ubuntu-latest
+
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'test-min') }}
+
+    env:
+      DRUPAL_TESTING_DRUPAL_VERSION: '~8.8.1'
+      DRUPAL_TESTING_COMPOSER_PROJECT_VERSION: 'dev-remove-core-recommended'
```
Comment (dbosen) [inline .github/workflows/test.yml]: Has been merged and released

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+          path: ~/.composer/cache
+          key: ${{ runner.os }}-composer-cache-${{ hashFiles('**/composer.json') }}
+          restore-keys: ${{ runner.os }}-composer-cache-
+
+      - name: Setup environment variables
+        run: |
+          echo "::add-path::$HOME/.composer/vendor/bin"
+
+      - name: Get build environment
+        run: composer global require thunder/drupal-testing:dev-feature/min-testing
```
Comment (dbosen) [inline .github/workflows/test.yml]: feature branch was merged and released.

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+          coverage: none
+          php-version: '7.3'
+          extensions: Imagick
+
+      - name: Setup environment variables
+        run: |
+          echo "::add-path::$HOME/.composer/vendor/bin"
+
+      - name: Get build environment
+        run: composer global require thunder/drupal-testing:dev-feature/min-testing
```
Comment (dbosen) [inline .github/workflows/test.yml]: again, the branch

File: modules/thunder_media/tests/src/Functional/FilenameTransliterationTest.php [hunk 8]
```diff
@@ -41,9 +41,14 @@ public function testFileTransliteration() {
     $account = $this->drupalCreateUser(['access site reports']);
     $this->drupalLogin($account);
 
-    $original = drupal_get_path('module', 'simpletest') . '/files';
-
-    \Drupal::service('file_system')->copy($original . '/image-1.png', PublicStream::basePath() . '/foo°.png');
+    if (file_exists('core/tests/fixtures/files/image-1.png')) {
+      \Drupal::service('file_system')->copy('core/tests/fixtures/files/image-1.png', PublicStream::basePath() . '/foo°.png');
```
Comment (dbosen) [inline modules/thunder_media/tests/src/Functional/FilenameTransliterationTest.php:45]: why the ° in the name?

Comment (chrfritsch) [reply modules/thunder_media/tests/src/Functional/FilenameTransliterationTest.php:45]: We are testing Transliteration here

File: .github/workflows/test.yml (hunk 2 above)
Comment (dbosen) [reply .github/workflows/test.yml]: but I am talking about drupal-testing

**Review by dbosen — APPROVED** [review]

**Pull request merged**